storage/memory/stats_buffer.go

```diff
@@ -43,7 +46,65 @@ func (self *StatsBuffer) Add(item *info.ContainerStats) {
 	self.buffer[self.index] = *item
 }
 
+// Returns up to maxResult elements in the specified time period (inclusive).
+// Results are from first to last. maxResults of -1 means no limit.
+func (self *StatsBuffer) InTimeRange(start, end time.Time, maxResults int) []*info.ContainerStats {
+
+	// NOTE: Since we store the elments in descending timestamp order "start" will
+	// be a higher index than "end".
+
+	var startIndex int
+	if start.IsZero() {
+		// None specified, start at the beginning.
+		startIndex = self.size - 1
```

Comment:
If the user doesn't report an end time it will set endIndex to 0 too, but good idea on just returning early if we're empty. Makes it much clearer. Added that.